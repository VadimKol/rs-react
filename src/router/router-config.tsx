import { createBrowserRouter, createRoutesFromElements, Route } from 'react-router-dom';

import { App } from '@/App';
import { ErrorPage } from '@/components/error/ErrorPage';
import { Main } from '@/components/main/Main';
import { NoMatch } from '@/components/no-match/NoMatch';

const routerConfig = createRoutesFromElements(
  <Route path="/" element={<App />} errorElement={<ErrorPage />}>
    <Route errorElement={<ErrorPage />}>
      <Route index element={<Main />} />
      <Route path="*" element={<NoMatch />} />
    </Route>
  </Route>,
);

export const router = createBrowserRouter(routerConfig);
