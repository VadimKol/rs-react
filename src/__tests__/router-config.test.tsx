import { render, waitFor } from '@testing-library/react';
import { RouterProvider } from 'react-router-dom';

import { router } from '@/router/router-config';

describe('router-config', () => {
  it('renders correctly', async () => {
    const { container } = render(<RouterProvider router={router} />);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
