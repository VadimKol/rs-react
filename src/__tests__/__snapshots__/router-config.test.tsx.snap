// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router-config renders correctly 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    />
  </header>
  <main
    class="main"
    role="presentation"
  >
    <section>
      <form
        class="search_form"
      >
        <input
          class="search"
          id="search"
          placeholder="Search..."
          type="text"
          value=""
        />
        <button
          aria-label="Search-button"
          class="search_button"
          type="submit"
        />
      </form>
    </section>
    <section
      class="results"
    >
      <div
        class="results_box"
      >
        <div
          class="not_found_box"
        >
          <div
            class="not_found"
          >
            <div
              class="image_container"
            >
              <div
                class="loader"
              />
              <img
                alt="Rick and Morty not found"
                class="character_img"
                src="stub-test"
                style="display: none;"
              />
            </div>
            <div
              class="text_container"
            >
              Characters not found
            </div>
          </div>
        </div>
      </div>
    </section>
  </main>
  <footer
    class="footer"
  >
    <div
      class="container"
    />
  </footer>
</div>
`;
